### ISSUE TITLE
FR: Expose the Passphrase for the "from-recovery-phrase" command

### ISSUE THREAD

**Issue by gitmachtl** [IntersectMBO/cardano-addresses#198]
Hi, after a little chat with @KtorZ i am opening this FR to expose the passphrase parameter via a cli parameter 

something like
``` console
$ cardano-address key from-recovery-phrase icarus "mnemonics" --passphrase "mySecretPassphrase"
```

Would be an awesome addition to the tool, thx!

**Comment by KtorZ** [comment]
CC @paweljakubas

**Comment by paweljakubas** [comment]
@teddyjfpender I am happy to add this functionality if you can put that in some future sprint

**Comment by paweljakubas** [comment]
hi @KtorZ So in essence we want to have:
1. ability to derive xprv with passwd as here https://github.com/input-output-hk/cardano-crypto/blob/04c211d7f9504b04eb6595cab62eb39a573467f1/src/Cardano/Crypto/Wallet.hs#L180
2. and then be able to "restore" xprv using this https://github.com/input-output-hk/cardano-crypto/blob/04c211d7f9504b04eb6595cab62eb39a573467f1/src/Cardano/Crypto/Wallet.hs#L171
(here newpasswd would be BS.empty)
3. and from user point of view password could be just text, or password hash (here we would use pbkdf2-hmac-sha512 to encode it as hex). 
Do I understand the request properly?
Many thanks!

**Comment by gitmachtl** [comment]
I am not familiar with the code you posted, but the requested passphrase was ment to be part of the pbkdf2 xprv key generation from a given mnemonics for the icarus method. Like its used for example in Trezor-HW-Wallets (exept for the 24 words with the entropy bug), like:

https://github.com/vacuumlabs/cardano-crypto.js/blob/8c8e467481380ba29db6295a9edbcdc54b1c63eb/features/key-derivation.js#L102

I have made a little [tool](https://github.com/gitmachtl/cardano-related-stuff/tree/master/cardano-mnemonic-rootkey) meanwhile that can do it for icarus, icarus-trezor and ledger method. 

Basically its about the password like ...
```
function generateIcarusMasterKey(seed, password) {
        const xprv = crypto.pbkdf2Sync(
                password,
                seed,
                4096,
                96,
                'sha512')

        xprv[0] &= 248
        xprv[31] &= 31
        xprv[31] |= 64

        return xprv;
}
```

**Comment by KtorZ** [comment]
@paweljakubas this 'passphrase' isn't to be mixed up the encryption passphrase. This is an additional passphrase that can be set during the master key generation as a second factor. This ensures that not only the recovery phrase is needed,  but also that extra passphrase,  in order to recover the root private key.

`cardano-wallet` exposes that feature in the API already and call it `mnemonic_second_factor` although weirdly enough, this is required to be a set of 9-12 BIP-0039 words as well instead of a plain UTF-8 string. I guess we didn't want people to mix that up with the spending, passphrase at the time and thought that a second factor as a mnemonic was good enough.

Note that this is only available for the new derivation style (used by Icarus and Shelley wallets) and _I think_,  Ledger's style also supports it.

**Comment by gitmachtl** [comment]
> _I think_, Ledger's style also supports it

Trezor supports it 1:1, Ledger uses another derivation style and also uses "mnemonic"+passphrase as the final password string to confuse devs even more 😆 https://github.com/LedgerHQ/orakolo/blob/0b2d5e669ec61df9a824df9fa1a363060116b490/src/python/orakolo/HDEd25519.py#L370-L387

**Comment by KtorZ** [comment]
Yes,  Trezor is the same style as Icarus / Shelley (except for 24 words 😔), and ledger has its own,  similar, but different. And..  Right,  I know remember how ledger had this weird things of concatenating the passphrase with an hard-coded "mnemonic"...  Sigh.

**Comment by paweljakubas** [comment]
@KtorZ @gitmachtl the PR addressing what you asked for (I hope so) is there https://github.com/input-output-hk/cardano-addresses/pull/202

**Comment by paweljakubas** [comment]
@gitmachtl @KtorZ PR https://github.com/input-output-hk/cardano-addresses/pull/202 merged - see READE.md in section 
```
How to generate a root private key with passphrase (root.xsk)
```
on instructions
